protocol/test/foundry/PipelineConvert.t.sol

```diff
@@ -0,0 +1,1461 @@
+        // then we pull our initial eth back out and we converted when we shouldn't have been able to (if we do in one tx)
+
+        // setup initial bean deposit
+        int96 stem = beanToLPDepositSetup(amount, users[1]);
+
+        // mint user 10 eth
+        uint256 ethAmount = 10e18;
+        MockToken(C.WETH).mint(users[1], ethAmount);
+
+        vm.prank(users[1]);
+        MockToken(C.WETH).approve(C.BEAN_ETH_WELL, ethAmount);
+
+        // add liquidity to well
+        uint256[] memory tokenAmountsIn = new uint256[](2);
+        tokenAmountsIn[0] = 0;
+        tokenAmountsIn[1] = ethAmount;
+
+        vm.prank(users[1]);
+        IWell(C.BEAN_ETH_WELL).addLiquidity(tokenAmountsIn, 0, users[1], type(uint256).max);
+
+        uint256 grownStalkBefore = bs.balanceOfGrownStalk(users[1], C.BEAN);
+
+        // convert beans to lp
+        beanToLPDoConvert(amount, stem, users[1]);
+
+        // it should be that we lost our grown stalk from this convert
+
+        uint256 grownStalkAfter = bs.balanceOfGrownStalk(users[1], C.BEAN_ETH_WELL);
+
+        assertTrue(grownStalkAfter == 0); // all grown stalk was lost
+        assertTrue(grownStalkBefore > 0);
+    }
+
+    // double convert uses up convert power so we should be left at no grown stalk after second convert
+    // (but still have grown stalk after first convert)
+    function testFlashloanManipulationLoseGrownStalkBecauseDoubleConvert(uint256 amount) public {
+        amount = bound(amount, 1000e6, 1000e6); // todo: update for range
+
+        // do initial pump update
+        updateMockPumpUsingWellReserves(C.BEAN_ETH_WELL);
+
+        // the main idea is that we start some positive deltaB, so a limited amount of converts are possible (1.2 eth worth)
+        // User One does a convert down, and that uses up convert power for this block
+        // someone adds more eth to the well, which means we're back too far over peg
+        // then User Two tries to do a convert down, but at that point the convert power has been used up, so they lose their grown stalk
+
+        // setup initial bean deposit
+        int96 stem = beanToLPDepositSetup(amount, users[1]);
+
+        // then setup a convert from user 2
+        int96 stem2 = beanToLPDepositSetup(amount, users[2]);
+
+        // if you deposited amount of beans into well, how many eth would you get?
+        uint256 ethAmount = IWell(C.BEAN_ETH_WELL).getSwapOut(IERC20(C.BEAN), IERC20(C.WETH), amount);
+
+        ethAmount = ethAmount.mul(12000).div(10000); // I need a better way to calculate how much eth out there should be to make sure we can swap and be over peg
+
+        addEthToWell(users[1], ethAmount);
+        
+        // go to next block
+        vm.roll(block.number + 1);
+
+        uint256 grownStalkBefore = bs.balanceOfGrownStalk(users[2], C.BEAN);
+
+        // update pump
+        updateMockPumpUsingWellReserves(C.BEAN_ETH_WELL);
+
+        convert.cappedReservesDeltaB(C.BEAN_ETH_WELL);
```

Comment:
this is not used.

Reply:
Removed here: https://github.com/BeanstalkFarms/Beanstalk/pull/840/commits/77024eaa74d6554e909e6501d7780df36f6a7d3e